### DISCUSSION TITLE
Incomplete object labeling can affect YOLO training?

### DISCUSSION THREAD

**Discussion by Josephts1** [ultralytics/ultralytics#18784] [Q&A]
Reading the section on collect and annotate your data for Ultralyticsm, more specifically the section on precision and accuracy. I am generating an algorithm for detecting mandarins in a specific environment and when reading the fragment on accuracy "Accuracy refers to how close the annotated data is to the true values", a doubt has arisen. Since there are these types of incomplete mandarin annotations (see image), can it affect the learning of the model? I emphasize that it is an object detection task (mandarins)

![image_2025-01-20_143530445](https://github.com/user-attachments/assets/ba6fc072-f967-4d26-9fd3-b885876763cd)
I appreciate your response

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Josephts1, thank you for your interest in Ultralytics 🚀! It's great to see your exploration into object detection tasks with Ultralytics models. For new users, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you'll find many helpful resources, including [data preparation guidelines](https://docs.ultralytics.com/datasets/) and [model training tips](https://docs.ultralytics.com/guides/model-training-tips/) that might address your concerns.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) for us to further investigate the issue.

If you have questions about custom training ❓, including challenges like object detection accuracy, we recommend sharing as much detail as possible. This includes your dataset examples, training logs, and how you’ve set up your annotations. Precise information will help us provide better guidance. You can also verify you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community to connect with other users and experts. For real-time assistance, check out [Discord](https://discord.com/invite/ultralytics) 🎧. Alternatively, explore more structured conversations on [Discourse](https://community.ultralytics.com/), or engage with others on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

As always, ensure you're using the latest version of `ultralytics` by upgrading the package and dependencies in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in a variety of environments, each pre-installed with necessary dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** Deep Learning AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. Continuous Integration (CI) tests validate the proper functioning of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to assist you quickly, but don't worry – an Ultralytics engineer will follow up soon to help clarify your specific question 😊✨!

**Comment by glenn-jocher** [comment]
@Josephts1 yes, incomplete or inaccurate annotations can negatively impact the training of your YOLO model, as they fail to provide the "ground truth" necessary for the model to learn effectively. For object detection tasks like yours, ensure that all mandarin instances are accurately and consistently labeled to improve model performance. You can find more detailed guidance on annotation best practices in the [Data Collection and Annotation guide](https://docs.ultralytics.com/guides/data-collection-and-annotation/).
